### PR TITLE
Legends to Typescript

### PR DESCRIPTION
**Pull request by ktw1016**

### CODE REVIEW

File: client/src/charts/legends/LegendList.tsx [hunk 1]
```diff
@@ -6,25 +6,20 @@ import { CheckBox, CheckBoxProps } from "src/components/CheckBox/CheckBox";
...
-  color?: string;
-}
+import "./LegendList.scss";
 
-interface LegendListProps {
+export interface LegendListProps {
   isHorizontal?: boolean;
-  items: LegendItemProps[];
+  items: LegendItemType[];
   onClick: Function;
```
Comment (Stephen-ONeil) [inline client/src/charts/legends/LegendList.tsx:16]: Haven't looked at this too closely, but I think this should be something like `onClick: (event: React.MouseEvent) => void`

Comment (Stephen-ONeil) [reply client/src/charts/legends/LegendList.tsx:16]: Or `onClick: React.ReactEventHandler<element>`?

File: client/src/components/CheckBox/CheckBox.tsx [hunk 2]
```diff
@@ -10,10 +11,10 @@ import "./CheckBox.scss";
 
 export interface CheckBoxProps {
   disabled?: boolean; // greys out checkbox and make it non-clickable if true
-  onClick: Function; // (id) => {}
-  id: string | number;
+  onClick?: Function; // (id) => {}
```
Comment (Stephen-ONeil) [inline client/src/components/CheckBox/CheckBox.tsx:14]: `(id: string | number) => void`?

File: client/src/components/CheckBox/CheckBox.tsx [hunk 3]
```diff
@@ -10,10 +11,10 @@ import "./CheckBox.scss";
...
+  id?: string | number;
   label: string; // required text for checkbox
-  active: boolean; // required, equivalent of 'checked'
+  active?: boolean; // required, equivalent of 'checked'
   color?: string; // defaults to primaryColor
   isSolidBox: boolean; // defaults to false
   container_style?: Object; // style for div, containing checkbox and label
```
Comment (Stephen-ONeil) [inline client/src/components/CheckBox/CheckBox.tsx:20]: Looks like this should be `React.CSSProperties`

**Review by Stephen-ONeil — CHANGES_REQUESTED** [review]

File: client/src/components/CheckBox/CheckBox.tsx (hunk 3 above)
Comment (ktw1016) [reply client/src/components/CheckBox/CheckBox.tsx:20]: Didn't know that existed!

**Review by Stephen-ONeil — APPROVED** [review]
:+1:

**Pull request merged**